spec/order_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'order'
+
+describe Order do
+  subject(:my_order) { described_class.new }
+
+  it 'can create an Order' do
+    expect(my_order).to be_a_kind_of(Order)
+  end
+
+  it 'has a default empty hash to store the order' do
+    expect(my_order.basket).to eq Hash.new(0)
+  end
+
+  describe '#add_dish' do
+    it 'adds an available dish to order' do
+      expect { my_order.add_dish('Pasta') }.to change { my_order.basket }.from(Hash.new(0)).to({ 'Pasta' => 12 })
+    end
+
+    it 'stores a dish in the order' do
+      my_order.add_dish('Pizza')
+      expect(my_order.basket).to include('Pizza')
+    end
+  end
+
+  describe '#remove_dish' do
+    it 'removes a dish from the order' do
+      my_order.add_dish('Burger')
```

Comment:
Since many methods require a dish to be added to the order for testing you could create a 'context' where all instances start with an order added. However I can see why you might not have decided to do that as your tests are very neatly organised and implementing that would require tests for different methods to be grouped into the same context rather than just all grouped by method.